translate.js

```diff
@@ -21,6 +24,11 @@ function versionToSemver (version) {
   if (version.indexOf('0.3.5-0') !== -1) {
     return '0.3.5';
   }
+  // This parses the obsolete nightly style where the date can have leading zeroes.
+  var nightly_parsed = version.match(/^([0-9]+\.[0-9]+\.[0-9]+)-nightly\.([0-9]+)\.0([1-9])\.0([1-9])(.*)$/);
```

Comment:
This does work for this one nightly but technically non-zero could also happen on the first position in the day or month.

I have just pushed https://github.com/ethereum/solc-bin/pull/103 and we might actually have some nightlies like that in there.

Reply:
Hm? This regex checks zeroes for months/days.

Reply:
Ah you mean **or**. Yes, you're correct, need to improve it for that.

Reply:
Could you run this on old nightlies? If there is no such instance, I think keeping this simple is better.

Reply:
Bad news. We have several 0.3.6 nightlies that need this:
| File name                                            | Reported version                                                 |
|------------------------------------------------------|------------------------------------------------------------------|
| `soljson-v0.3.6-nightly.2016.8.27+commit.91d4fa47.js`| `0.3.6-nightly.2016.08.27+commit.91d4fa47.Emscripten.clang`      |
| `soljson-v0.3.6-nightly.2016.8.27+commit.91d4fa47.js`| `0.3.6-nightly.2016.08.27+commit.91d4fa47.Emscripten.clang`      |
| `soljson-v0.3.6-nightly.2016.8.29+commit.b8060c55.js`| `0.3.6-nightly.2016.08.29+commit.b8060c55.Emscripten.clang`      |
| `soljson-v0.3.6-nightly.2016.8.29+commit.b8060c55.js`| `0.3.6-nightly.2016.08.29+commit.b8060c55.Emscripten.clang`      |
| `soljson-v0.3.6-nightly.2016.8.30+commit.cf974fd1.js`| `0.3.6-nightly.2016.08.30+commit.cf974fd1.Emscripten.clang`      |
| `soljson-v0.3.6-nightly.2016.8.30+commit.cf974fd1.js`| `0.3.6-nightly.2016.08.30+commit.cf974fd1.Emscripten.clang`      |
| `soljson-v0.3.6-nightly.2016.8.31+commit.3ccd1986.js`| `0.3.6-nightly.2016.08.31+commit.3ccd1986.Emscripten.clang`      |
| `soljson-v0.3.6-nightly.2016.8.31+commit.3ccd1986.js`| `0.3.6-nightly.2016.08.31+commit.3ccd1986.Emscripten.clang`      |

I have also detected another variant that triggers a crash:
| File name                                            | Reported version                                                 |
|------------------------------------------------------|------------------------------------------------------------------|
| `soljson-v0.3.4-nightly.2016.6.6+commit.e97ac4fb.js` | `0.3.4-0/Release-Emscripten/clang/Interpreter`                   |
| `soljson-v0.3.4-nightly.2016.6.6+commit.e97ac4fb.js` | `0.3.4-0/Release-Emscripten/clang/Interpreter`                   |
| `soljson-v0.3.4-nightly.2016.6.8+commit.093790d7.js` | `0.3.4-0/Release-Emscripten/clang/Interpreter`                   |
| `soljson-v0.3.4-nightly.2016.6.8+commit.093790d7.js` | `0.3.4-0/Release-Emscripten/clang/Interpreter`                   |
| `soljson-v0.3.4-nightly.2016.6.8+commit.ccddd6fd.js` | `0.3.4-0/Release-Emscripten/clang/Interpreter`                   |
| `soljson-v0.3.4-nightly.2016.6.8+commit.ccddd6fd.js` | `0.3.4-0/Release-Emscripten/clang/Interpreter`                   |
| `soljson-v0.3.4-nightly.2016.6.8+commit.d593166d.js` | `0.3.4-0/Release-Emscripten/clang/Interpreter`                   |
| `soljson-v0.3.4-nightly.2016.6.8+commit.d593166d.js` | `0.3.4-0/Release-Emscripten/clang/Interpreter`                   |

Reply:
The `0.3.4-0` case is partially handled, for `0.1.3-0` and `0.3.5-0`. For both it was understood that the commit hash was set to 0.

However based on your comment it seems that happened for nightlies only? Can you check 0.1.3 and 0.3.4 what they output?

Reply:
@cameel you may have missed these follow up questions

Reply:
Oh sorry, I was sure I answered everything here. Apparently missed your question about releases.

Releases obviously do not have this problem because they do not include a date. You can check that on the list I prepared in the past when I was rebuilding binaries: https://github.com/ethereum/solc-bin/pull/64#issuecomment-711221554